rust/src/ffiutil.rs

```diff
@@ -120,6 +121,17 @@ pub(crate) fn ref_from_raw_ptr<T>(p: *mut T) -> &'static mut T {
 // return a Result (using the std Error).
 // TODO: Try upstreaming this into the glib crate?
 
+/// Convert C results (int + GError convention) to anyhow.
+pub(crate) fn int_gerror_to_result(res: i32, gerror: *mut glib_sys::GError) -> anyhow::Result<()> {
+    if res != 0 {
+        Ok(())
+    } else {
+        anyhow::ensure!(!gerror.is_null(), "unknown failure (NULL Gerror)");
```

Comment:
I'd lean towards aborting here (i.e. `panic!()`) since it's a bad bug if it happens, and it really shouldn't - it's definitely not a "user error".

Reply:
Ack, I've turned this one into an `assert!()`,